plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1206,6 +1214,175 @@ def is_valid_address(cls, address: Address) -> bool:
+        tx: Any,
+        eth_value: int = 0,
+        gas: Optional[int] = None,
+        gas_price: Optional[int] = None,
+        max_fee_per_gas: Optional[int] = None,
+        max_priority_fee_per_gas: Optional[int] = None,
+    ) -> Optional[JSONLike]:
+        """Build transaction method."""
+        nonce = self.api.eth.get_transaction_count(sender_address)
+        tx_params = {
+            "nonce": nonce,
+            "value": eth_value,
+        }
+        if gas is not None:
+            tx_params["gas"] = gas
+        if gas_price is not None:
+            tx_params["gasPrice"] = gas_price
+        if max_fee_per_gas is not None:
+            tx_params["maxFeePerGas"] = max_fee_per_gas  # pragma: nocover
+        if max_priority_fee_per_gas is not None:
+            tx_params[
+                "maxPriorityFeePerGas"
+            ] = max_priority_fee_per_gas  # pragma: nocover
+        if (
+            gas_price is None
+            and max_fee_per_gas is None
+            and max_priority_fee_per_gas is None
+        ):
+            gas_data = self.try_get_gas_pricing()
+            if gas_data:
+                tx_params.update(gas_data)  # pragma: nocover
+        tx = tx.buildTransaction(tx_params)
+        return tx
+
+    def get_transaction_transfer_logs(  # pylint: disable=too-many-arguments,too-many-locals
+        self,
+        contract_interface: Dict[str, str],
+        tx_hash: str,
+        target_address: Optional[str] = None,
+    ) -> JSONLike:
+        """
+        Get all transfer events derived from a transaction.
+
+        :param contract_interface: the contract interface
+        :param tx_hash: the transaction hash
+        :param target_address: optional address to filter tranfer events to just those that affect it
+        :return: the transfer logs
+        """
+        contract = self.get_contract_instance(contract_interface)
+
+        try:
+            tx_receipt = self.api.eth.get_transaction_receipt(tx_hash)
+            if tx_receipt is None:
+                raise ValueError  # pragma: nocover
+
+        except (TransactionNotFound, ValueError):  # pragma: nocover
+            return dict(logs=[])
+
+        # Due to serialization, event topics must be converted again to HexBytes or processReceipt will fail
+        tx_receipt = rebuild_receipt(tx_receipt)
+
+        transfer_logs = contract.events.Transfer().processReceipt(tx_receipt)
+
+        transfer_logs = [
+            {
+                "from": log["args"]["from"],
+                "to": log["args"]["to"],
+                "value": log["args"]["value"],
+                "token_address": log["address"],
+            }
+            for log in transfer_logs
+        ]
+
+        if target_address:
+            transfer_logs = list(
+                filter(
+                    lambda log: target_address in (log["from"], log["to"]),  # type: ignore
+                    transfer_logs,
+                )
+            )
```

Comment:
This part is contract specific, so should be done in the contract from which this is called from imo